tests/Medidata.ZipkinTracer.Core.Test/TraceProviderTests.cs

```diff
@@ -218,5 +218,91 @@ public void GetNext()
             Assert.AreEqual(sut.SpanId, nextTraceProvider.ParentSpanId);
             Assert.AreEqual(sut.IsSampled, nextTraceProvider.IsSampled);
         }
+
+        [TestMethod]
+        public void IdsWithLessThan32Characters()
+        {
+            // Arrange
+            var fixture = new Fixture();
+            var traceId = "48485a3953bb612";
+            var spanId = "48485a3953bb613";
+            var parentSpanId = "48485a3953bb614";
+            var isSampled = fixture.Create<bool>();
+
+            var context = GenerateContext(traceId, spanId, parentSpanId, isSampled);
+
+            // Act
+            var sut = new TraceProvider(new ZipkinConfig(), context);
+
+            // Assert
+            Assert.AreEqual(traceId, sut.TraceId);
+            Assert.AreEqual(spanId, sut.SpanId);
+            Assert.AreEqual(parentSpanId, sut.ParentSpanId);
+        }
+
+        [TestMethod]
+        public void IdsWith32Characters()
+        {
+            // Arrange
+            var fixture = new Fixture();
+            var traceId = "48485a3953bb6124";
+            var spanId = "48485a3953bb6125";
+            var parentSpanId = "48485a3953bb6126";
+            var isSampled = fixture.Create<bool>();
+
+            var context = GenerateContext(traceId, spanId, parentSpanId, isSampled);
+
+            // Act
+            var sut = new TraceProvider(new ZipkinConfig(), context);
+
+            // Assert
+            Assert.AreEqual(traceId, sut.TraceId);
+            Assert.AreEqual(spanId, sut.SpanId);
+            Assert.AreEqual(parentSpanId, sut.ParentSpanId);
+        }
+
+        [TestMethod]
+        public void IdsWith64Characters()
+        {
+            // Arrange
+            var fixture = new Fixture();
+            var traceIdLower16Chars = "48485a3953bb6124";
+            var traceId = "18485a3953bb6124" + traceIdLower16Chars;
+            var spanIdLower16Chars = "48485a3953bb6125";
```

Comment:
please only consider the trace id, not the span or parent span ids, as spans within a trace are always 64-bit